File: backend/src/schemas/audit.py
from datetime import datetime
from typing import Any, Dict, List, Optional
from uuid import UUID
from pydantic import BaseModel, Field


class AuditLogResponse(BaseModel):
    """Schema for audit log entry responses."""
    id: UUID
    user_id: str
    event_type: str
    task_id: UUID
    task_data: Dict[str, Any]
    event_id: str
    timestamp: datetime

    class Config:
        from_attributes = True


class AuditLogListResponse(BaseModel):
    """Paginated audit log response."""
    items: List[AuditLogResponse]
    total: int
    offset: int
    limit: int

File: frontend/types/audit.ts
export interface AuditLogEntry {
  id: string
  user_id: string
  event_type: 'created' | 'updated' | 'completed' | 'deleted'
  task_id: string
  task_data: Record<string, any>
  event_id: string
  timestamp: string
}

export interface AuditLogListResponse {
  items: AuditLogEntry[]
  total: number
  offset: number
  limit: number
}
